src/miner/miner.rs

```diff
@@ -138,4 +141,56 @@ mod test {
 
     Ok(())
   }
+  #[test]
+  ///The worker should return a challenge with expected difficulty not realized difficulty.
+  fn worker_returns_challenge_with_expected_difficulty() -> Result<(), String> {
+    let miner = Miner::default();
+    let block_id: Vec<u8> = b"1111111111111111".iter().copied().collect();
+    let peer_id: Vec<u8> = b"1111111111111111".iter().copied().collect();
+
+    let timestamp: f64 = utc_seconds_f64();
+
+    let challenge: Challenge = Challenge {
+      difficulty: 2,
+      timestamp,
+      block_id: block_id.clone(),
+      peer_id: peer_id.clone(),
+    };
+
+    miner.worker.send(challenge.clone());
+    while None == miner.try_create_consensus() {}
+
+    let consensus: BlockConsensus;
+    //the second answer if any is guaranteed to be higher than the expected difficulty but the expected diff should stay the same
```

Comment:
I see assertions about the state of realized difficulty and then comparing two instances of realized difficulty but nothing asserting on the state of expected difficulty. How can we check that? It sounds like we need one more assertion to make sure expected difficulty did not change.

Reply:
"expected difficulty" is what goes as input, in this case `challenge.difficulty`.